controllers/doguVolumeManager.go

```diff
@@ -272,3 +276,98 @@ func isPvcResizeApplicable(pvc *corev1.PersistentVolumeClaim) bool {
 	}
 	return false
 }
+
+type dataVolumeSizeStep struct {
+	client        client.Client
+	eventRecorder record.EventRecorder
+}
+
+func (d *dataVolumeSizeStep) GetStartCondition() string {
+	return startConditionValidate
+}
+
+// Execute executes the step and returns the next state and if the step fails an error.
+// The error can be a requeueable error so that the step will be executed again.
+func (d *dataVolumeSizeStep) Execute(ctx context.Context, dogu *doguv2.Dogu) (string, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Start Validate Volume Size..")
+	pvc, err := dogu.GetDataPVC(ctx, d.client)
+	if err != nil {
+		return "", err
+	}
+	currentSize := pvc.Status.Capacity.Storage()
+	minDataSize, err := dogu.GetMinDataVolumeSize()
+	if err != nil {
+		logger.Error(err, "failed to get min data volume size")
+		return "", err
+	}
+	if minDataSize.Value() > currentSize.Value() {
+		logger.Info("resize not finished yet... requeue")
+		return "", notResizedError{
+			state:       d.GetStartCondition(),
+			requeueTime: time.Minute * 1,
+		}
+	}
+
+	err = SetCurrentDataVolumeSize(ctx, d.client, dogu)
+
+	if err != nil {
+		return "", err
+	}
+
+	// Finish Resizing
+	err = dogu.ChangeRequeuePhaseWithRetry(ctx, d.client, "")
+	if err != nil {
+		return "", err
+	}
+
+	return async.FinishedState, nil
+}
+
+// SetCurrentDataVolumeSize set the current DataVolumeSize within the status of the dogu
+func SetCurrentDataVolumeSize(ctx context.Context, client client.Client, doguResource *doguv2.Dogu) error {
```

Comment:
The function and prodecures of the callers are not clear.

This step queries the pvc twice if the change was successful.
I would extract the logic that fetches the pvc and checks that: actualPVC >= specSize.
With this information you can call this function with the condition parameter (true|false)